### PR TITLE
Add date_trunc to scalar functions to support date_trunc during ingestion

### PR DESCRIPTION
**Pull request by npawar**
date_trunc was only available during query time. Adding it to scalar functions, so that it can be used during ingestion.
All logic is simply moved from DateTruncTransformFunction class to shared utils. No new logic has been introduced.

Also as part of this PR, split the InbuiltFunctionsTest into test files of their own, per function group (arithmetic, array, json, datetime), because that file was getting too long.

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/function/DateTimeUtils.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+/**
+ * Helper methods and constructs for datetrunc function
+ */
+public class DateTimeUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DateTimeUtils.class);
+  private static final DateTimeFieldType QUARTER_OF_YEAR = new QuarterOfYearDateTimeField();
+
+  public static DateTimeField getTimestampField(ISOChronology chronology, String unitString) {
+    switch (unitString) {
```
Comment (fx19880617) [inline pinot-common/src/main/java/org/apache/pinot/common/function/DateTimeUtils.java:44]: Do we need to make this case insensitive?

Comment (fx19880617) [reply pinot-common/src/main/java/org/apache/pinot/common/function/DateTimeUtils.java:44]: Since this could also be used in the ingestion functions

Comment (npawar) [reply pinot-common/src/main/java/org/apache/pinot/common/function/DateTimeUtils.java]: Done

**Review by fx19880617 — APPROVED** [review]

**Comment by fx19880617** [comment]
Maybe also change the PR title to mention that we move date_trunc to scalar function.
Something like: Add date_trunc to scalar functions to support date_trunc during ingestion

**Pull request merged**